File: index-manager/internal/domain/health.go
package domain

// HealthStatus represents the health status of the service
type HealthStatus struct {
	Status    string            `json:"status"` // healthy, unhealthy, degraded
	Version   string            `json:"version"`
	Checks    map[string]string `json:"checks"`
	Timestamp string            `json:"timestamp"`
}

File: ml-framework/nc_ml/health.py
"""Health check aggregation logic."""


def aggregate_health_status(checks: dict[str, bool]) -> str:
    """Aggregate individual check results into an overall status.

    - All True or empty -> "healthy"
    - Some False -> "degraded"
    - All False -> "unhealthy"
    """
    if not checks:
        return "healthy"

    values = list(checks.values())
    if all(values):
        return "healthy"
    if not any(values):
        return "unhealthy"
    return "degraded"
